pkg/storagev2/inmemchunkstore/transaction.go

```diff
@@ -81,19 +81,14 @@ func (s *TxChunkStore) Delete(ctx context.Context, addr swarm.Address) error {
 
 // Commit implements the Tx interface.
 func (s *TxChunkStore) Commit() error {
-	if err := s.IsDone(); err != nil {
-		return err
-	}
-	s.TxState.Done()
-	return nil
+	return s.TxState.Done()
 }
 
 // Rollback implements the Tx interface.
 func (s *TxChunkStore) Rollback() error {
-	if err := s.IsDone(); err != nil {
+	if err := s.TxState.Done(); err != nil {
 		return err
 	}
-	defer s.TxState.Done()
 
 	s.opsMu.Lock()
```

Comment:
Same, I dont think we need this mutex Lock anymore right?

Reply:
Fixed.